src/nodes.coffee

```diff
@@ -1532,6 +1532,9 @@ exports.Param = class Param extends Base
         obj = obj.variable
       # * assignments within destructured parameters `{foo:bar}`
       if obj instanceof Assign
+        # ... possibly with a default value
+        if obj.value instanceof Assign
```

Comment:
No, I was just wondering if this nesting of checks were going to break this one. We should probably have tests for this behavior, though, if it's meant to exist...